### PR TITLE
Throw when parsedUrl is not provided

### LINKED ISSUE #990: Cannot read property 'replace' of undefined
I tried to upgrade from next@2.0.0-beta.20 but something changed in the beta.21 specifically, I get the stack trace below when running `yarn dev`. Here is my app: https://github.com/relatenow/relate

<details>
TypeError: Cannot read property 'replace' of undefined
    at toRoute (/Users/sdubois/Documents/relate/node_modules/next/dist/lib/router/router.js:618:14)
    at new Router (/Users/sdubois/Documents/relate/node_modules/next/dist/lib/router/router.js:85:19)
    at renderPage (/Users/sdubois/Documents/relate/node_modules/next/dist/server/render.js:143:25)
    at /Users/sdubois/Documents/relate/node_modules/next/dist/server/document.js:56:22
    at renderStatic (/Users/sdubois/Documents/relate/node_modules/glamor/lib/server.js:18:14)
    at Function.getInitialProps (/Users/sdubois/Documents/relate/node_modules/next/dist/server/document.js:55:45)
    at _callee$ (/Users/sdubois/Documents/relate/node_modules/next/dist/lib/utils.js:36:30)
    at tryCatch (/Users/sdubois/Documents/relate/node_modules/regenerator-runtime/runtime.js:64:40)
    at Generator.invoke [as _invoke] (/Users/sdubois/Documents/relate/node_modules/regenerator-runtime/runtime.js:355:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/sdubois/Documents/relate/node_modules/regenerator-runtime/runtime.js:116:21)
(node:4021) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'replace' of undefined
</details>

### PR DESCRIPTION
**Pull request by timneutkens**
Fixes #990

### CODE REVIEW

**Comment by timneutkens** [comment]
cc @arunoda

**Comment by arunoda** [comment]
We should parse this. But we should throw if the user provided something other than a parsedUrl as the third argument.

**Comment by timneutkens** [comment]
@arunoda This throws in all cases where `parsedUrl.query` doesn't exist, which is fine since passing it through without `true` as second parameter will cause a re-parse otherwise, which is unneeded right? Or do you want to keep the url parsing when passing only `req, res` instead of `req, res, parsedUrl`?

**Comment by arunoda** [comment]
If the parsedUrl is undefined or null, I need to do parse the URL ourselves.

**Comment by timneutkens** [comment]
@arunoda fixed 🙌

File: server/index.js [hunk 1]
```diff
@@ -121,7 +125,11 @@ export default class Server {
 
   async start (port) {
     await this.prepare()
-    this.http = http.createServer(this.getRequestHandler())
+    const handle = this.getRequestHandler()
+    this.http = http.createServer((req, res) => {
+      const parsedUrl = parse(req.url, true)
```
Comment (arunoda) [inline server/index.js]: I think we don't need this line.

**Comment by arunoda** [comment]
Thanks @timneutkens 
This looks great.

**Pull request merged**